org.eclipse.cdt.cross.arc.gnu/src/org/eclipse/cdt/cross/arc/gnu/common/ResourceChangeListener.java

```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * This program and the accompanying materials are made available under the terms of the Common
+ * Public License v1.0 which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/cpl-v10.html
+ *
+ * Copyright (c) 2016 Synopsys, Inc.
+ *******************************************************************************/
+
+package org.eclipse.cdt.cross.arc.gnu.common;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IResourceChangeEvent;
+import org.eclipse.core.resources.IResourceChangeListener;
+import org.eclipse.core.resources.IResourceDelta;
+
+/**
+ * This class is intended for tracking the resource's changes.
+ */
+public class ResourceChangeListener implements IResourceChangeListener {
```

Comment:
As discussed offline, please reconsider the need for this class - it looks like there a listener, whose main purpose is to pass event down to another listener. This looks awkward, you probably can have just one listener class for now, and then you can split them sometime in the future, should the need arise.